src/cljs/athens/electron.cljs

```diff
@@ -199,9 +203,136 @@
     (fn [db _]
       (:db/remote-graph db)))
 
+  ;; ------- db -picker related implementation--------------------------------
+
+  ;; Subs
+
+  (reg-sub
+    :db-picker/all-dbs
+    (fn [db _]
+      (:db-picker/all-dbs db)))
+
+  ;; helper functions
+
+  (defn get-db-name
+    "From a dbpath find out db name.
+     e.g : /home/sid597/Desktop/athens db/test db/index.transit --> test db"
+    [dbpath]
+    (.basename path (.dirname path dbpath)))
+
+  (defn check-duplicate-db
+    "Check if a db is already in the list.
+     Implementation thoughts
+      - Should this be done whenever a new db is added or should we maintain a
+        set universally and check that?
+      - No performance issues if this is done on every new db addition because
+        not many dbs would be made frequently to cause performance issue. What
+        about merging 2 large dbs? Not there yet, but in that case also calculating
+        the set of all paths is O(n).
+      - Also re-frame is in-memory so saving data which can be easily calculated is
+        not a good idea I think"
+    [db-list check-path]
+    (contains? (into #{} (map #(:path %) db-list)) check-path))
 
   ;; Events
 
+  (reg-event-fx
+    :db-picker/add-new-db
+    (fn [{:keys [db]} [_ dbpath]]
+      "Add a new to db picker list.
+       Can be invoked when a db is added, opened, moved, selected or removed.
+       Update local storage value with the new db list"
+      (let [current-db-list (:db-picker/all-dbs db)
+            duplicate?       (check-duplicate-db
+                               current-db-list
+                               dbpath)]
+        (if duplicate?
+          {:dispatch [:alert/js (str "Database already in the list")]}
+          (let [dbname  (get-db-name dbpath)
+                newdb   {:path dbpath
+                         :name dbname}
+                all-dbs (conj current-db-list newdb)]
+            {:db       (assoc db :db-picker/all-dbs all-dbs)
+             :dispatch [:local-storage/set-db-picker-list]})))))
```

Comment:
With recent `re-frame` updates it's recommended to use `:fx` instead of `:dispatch`.

https://github.com/Day8/re-frame/blob/master/docs/Effects.md#the-effects-map

Reply:
@neotyk Ok I did not know that, in the website docs they are [still using](https://day8.github.io/re-frame/EffectfulHandlers/#step-1-of-plan) `:dispatch`